map/src/main/kotlin/edu/artic/map/MapFragment.kt

```diff
@@ -250,9 +265,40 @@ class MapFragment : BaseViewModelFragment<MapViewModel>() {
         mapView.onLowMemory()
     }
 
+    private fun loadAmenities(amenitiesList: List<MapItem.Annotation>) {
+        amenitiesMarkerList
+                .forEach { marker -> marker.remove() }
+
+        amenitiesList.forEach { mapItem ->
+            amenitiesMarkerList.add(
+                    map.addMarker(MarkerOptions()
+                            .position(mapItem.item.toLatLng())
+                            .zIndex(0f)
+                    )
+
+            )
+        }
+    }
+
+    private fun loadSpaceOrLandmarks(spaceOrLandmarkList: List<MapItem.Annotation>) {
+        spaceOrLandmarkMarkerList
+                .forEach { marker -> marker.remove() }
+
+        spaceOrLandmarkList.forEach { mapItem ->
```

Comment:
Ive seen this code repeated in multiple places doing the exact same thing save a small change in the `BitmapDescriptorFactory`. please consider creating an intermediate `inline` method that does all the logic for galleryMarkers, spaceOrLandmarkerList, etc. the same way:
```
private inline fun loadMarkers(list: List<MapItem.Annotation>, markerBuilder: (MapItem.Annotation) -> Marker) {
  list.forEach {  it.remove () }
  list.clear()
  list.add(map.addMarker(markerBuilder(mapItem)))
}
```

Reply:
I guess there can be an inline method for this...kinda

Reply:
Created it and used where it makes sense (basically amentities and landmarkAndSpace